### ISSUE TITLE
`restic copy -o local.connections=1` stuck forever.

### ISSUE THREAD

**Issue by DRON-666** [restic/restic#3897]
Output of `restic version`
--------------------------
restic 0.14.0 compiled with go1.19 on windows/amd64

How did you run restic exactly?
-------------------------------
`restic -r repo2 copy --from-repo repo1 -o local.connections=1 latest`

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
local backend

Expected behavior
-----------------
```
[0:00] 100.00%  2 / 2 packs copied
snapshot d916e6d0 saved
```
Actual behavior
---------------
restic stuck forever on:
```
[6:38] 0.00%  0 / 2 packs copied
```
Steps to reproduce the behavior
-------------------------------
```
C:\copy-test>restic -r repo1 init
debug log file debug.log
debug enabled
created restic repository 02e1653c12 at repo1

Please note that knowledge of your password is required to access
the repository. Losing your password means that your data is
irrecoverably lost.

C:\copy-test>restic -r repo1 backup somedata
debug log file debug.log
debug enabled
repository 02e1653c opened (repository version 2) successfully, password is correct
created new cache in C:\Users\DRON\AppData\Local\restic
no parent snapshot found, will read all files

Files:           1 new,     0 changed,     0 unmodified
Dirs:            0 new,     0 changed,     0 unmodified
Added to the repository: 8.850 MiB (3.326 KiB stored)

processed 1 files, 781.250 MiB in 0:02
snapshot fe8c4b37 saved

C:\copy-test>restic -r repo2 init
debug log file debug.log
debug enabled
created restic repository b305c633e9 at repo2

Please note that knowledge of your password is required to access
the repository. Losing your password means that your data is
irrecoverably lost.

C:\copy-test>restic -r repo2 copy --from-repo repo1 -o local.connections=1 latest
debug log file debug.log
debug enabled
repository 02e1653c opened (repository version 2) successfully, password is correct
repository b305c633 opened (repository version 2) successfully, password is correct
created new cache in C:\Users\DRON\AppData\Local\restic

snapshot fe8c4b37 of [C:\copy-test\somedata] at 2022-08-28 02:44:31.3080169 +0300 MSK)
  copy started, this may take a while...
[6:38] 0.00%  0 / 2 packs copied
```
[debug.log](https://github.com/restic/restic/files/9438844/debug.log)

Do you have any idea what may have caused this?
-----------------------------------------------
No.

Do you have an idea how to solve the issue?
-------------------------------------------
No

**Comment by DRON-666** [comment]
I was able to bisect it to https://github.com/restic/restic/commit/e597b99b55d0f00a23340f5b09961cdc269623e6: `repackWorkerCount` equal to zero [here](https://github.com/restic/restic/blob/c169e37139db8be743ba4f2ef24cf465dc1bc597/internal/repository/repack.go#L116).

**Comment by MichaelEischer** [comment]
Thanks for the bugreport. This should be fixed by #3898.
